src/features/workspaces/workspace-detail/WorkspaceDetail.tsx

```diff
@@ -70,15 +92,31 @@ export const WorkspaceDetail = () => {
 
   useEffect(() => {
     if (activeTabString === 'roles' && enableRoles) {
-      fetchGroupsData(groupsPage, groupsPerPage);
+      fetchGroupsData(groupsPage, groupsPerPage, groupsSortBy, groupsDirection, groupsFilters);
     }
-  }, [fetchGroupsData, groupsPage, groupsPerPage, activeTabString, enableRoles]);
+  }, [fetchGroupsData, groupsPage, groupsPerPage, groupsSortBy, groupsDirection, groupsFilters, activeTabString, enableRoles]);
 
   const handleGroupsPaginationChange = useCallback((page: number, perPage: number) => {
     setGroupsPage(page);
     setGroupsPerPage(perPage);
   }, []);
 
+  const handleGroupsSort = useCallback((_event: React.MouseEvent | React.KeyboardEvent, key: string, direction: 'asc' | 'desc') => {
+    setGroupsSortBy(key);
+    setGroupsDirection(direction);
+    setGroupsPage(1); // Reset to first page when sorting
+  }, []);
+
+  const handleGroupsSetFilters = useCallback((newFilters: Partial<{ name: string }>) => {
```

Comment:
instead of these callback methods, we might want to use the `useDataViewFilters` method that we use with our other DataView tables. Like in this table: https://github.com/RedHatInsights/insights-rbac-ui/blob/master/src/features/roles/RolesTable.tsx#L103